tfx_bsl/beam/run_inference_test.py

```diff
@@ -638,12 +641,120 @@ def test_request_serialized_example(self):
             model_name='test_model',
             version_name='test_version',
             use_serialization_config=True))
-    remote_predict = run_inference._RemotePredictDoFn(inference_spec_type, None)
+    remote_predict = run_inference._RemotePredictDoFn(
+      None, fixed_inference_spec_type=inference_spec_type)
+    remote_predict._setup_model(remote_predict._fixed_inference_spec_type)
     result = list(remote_predict._prepare_instances([example]))
     self.assertEqual(result, [{
         'b64': base64.b64encode(example.SerializeToString()).decode()
     }])
 
 
+class RunInferenceCoreTest(RunInferenceFixture):
+
+  def _build_keras_model(self, add):
+    """Builds a dummy keras model with one input and output."""
+    inp = tf.keras.layers.Input((1,), name='input')
+    out = tf.keras.layers.Lambda(lambda x: x + add)(inp)
+    m = tf.keras.models.Model(inp, out)
+    return m
+
+  def _new_model(self, model_path, add):
+    """Exports a keras model in the SavedModel format."""
+    class WrapKerasModel(tf.keras.Model):
+      """Wrapper class to apply a signature to a keras model."""
+      def __init__(self, model):
+        super().__init__()
+        self.model = model
+
+      @tf.function(input_signature=[
+          tf.TensorSpec(shape=[None], dtype=tf.string, name='inputs')
+      ])
+      def call(self, serialized_example):
+        features = {
+          'input': tf.compat.v1.io.FixedLenFeature(
+            [1],
+            dtype=tf.float32,
+            default_value=0
+          )
+        }
+        input_tensor_dict = tf.io.parse_example(serialized_example, features)
+        return self.model(input_tensor_dict)
+
+    model = self._build_keras_model(add)
+    wrapped_model = WrapKerasModel(model)
+    tf.compat.v1.keras.experimental.export_saved_model(
+      wrapped_model, model_path, serving_only=True
+    )
+    return self._get_saved_model_spec(model_path)
+
+  def _decode_value(self, pl):
+    """Returns output value from prediction log."""
+    out_tensor = pl.predict_log.response.outputs['output_1']
```

Comment:
Where does 'output_1' come from? If it's output tensor name, consider to set this up in build_keras_model.

Reply:
Yes this is just the default output tensor name. Do you know how to configure this? I'm using:

```py
@tf.function(input_signature=[
  tf.TensorSpec(shape=[None], dtype=tf.string, name='inputs')
])
```

to configure the input signature but there is no equivalent `output_signature` argument.

Reply:
You can set it in keras layer:
out = tf.keras.layers.Lambda(lambda x: x + add, name='output1')(inp)

Reply:
Following up on this, it turns out the `name` attribute on a keras layer isn't preserved in the signature (because we end up wrapping the model with `WrapKerasModel`. However I was able to find in the documentation that you can return a dictionary from a `@tf.function` like:

```py
@tf.function(...)
def my_func(x):
  val = ...
  return {'output': val}
```

and this name is preserved in the output signature.

(see: https://www.tensorflow.org/guide/saved_model#specifying_signatures_during_export)